test/sass/functions_test.rb

```diff
@@ -1419,8 +1439,10 @@ def test_map_values_checks_type
 
   def test_map_has_key
     assert_equal "true", evaluate("map-has-key((foo: 1, bar: 1), foo)")
+    assert_equal "true", evaluate("map-has-key((foo: (bar: 1)), foo, bar)")
     assert_equal "false", evaluate("map-has-key((foo: 1, bar: 1), baz)")
     assert_equal "false", evaluate("map-has-key((), foo)")
+    assert_equal "false", evaluate("map-has-key((foo: (bar: 1)), foo, bing)")
```

Comment:
Test nesting deeper than actually exists, nesting through values that aren't maps, and nesting through empty maps.